crates/kornia-io/src/functional.rs

```diff
@@ -132,6 +132,385 @@ pub fn read_image_any_rgb8(file_path: impl AsRef<Path>) -> Result<Image<u8, 3>,
+/// assert_eq!(image.rows(), 195);
+/// assert_eq!(image.num_channels(), 3);
+/// ```
+pub fn decode_image_bytes_rgb8(image_data: &[u8], format_hint: Option<&str>) -> Result<Image<u8, 3>, IoError> {
+    // For JPEG data, use turbojpeg if the feature is enabled
+    #[cfg(feature = "turbojpeg")]
+    if format_hint.map_or(false, |fmt| fmt.eq_ignore_ascii_case("jpg") || fmt.eq_ignore_ascii_case("jpeg")) 
+        || image::guess_format(image_data).map_or(false, |fmt| fmt == image::ImageFormat::Jpeg) {
+        return decode_image_jpegturbo_rgb8(image_data);
+    }
+
+    // Use the image crate for other formats
+    let image_reader = image::load_from_memory(image_data)
+        .map_err(|e| IoError::ImageDecodingError(e.to_string()))?;
+    
+    let rgb_image = image_reader.to_rgb8();
+
+    let (width, height) = rgb_image.dimensions();
+    let size = ImageSize {
+        width: width as usize,
+        height: height as usize,
+    };
+
+    let image = Image::new(size, rgb_image.into_raw())?;
+
+    Ok(image)
+}
+
+#[cfg(feature = "turbojpeg")]
+/// Decodes a JPEG image in `GRAY8` format directly from bytes.
+///
+/// The method decodes the JPEG image data from the provided byte array leveraging the libjpeg-turbo library.
+///
+/// # Arguments
+///
+/// * `jpeg_data` - The raw JPEG data as a byte array.
+///
+/// # Returns
+///
+/// An image containing the decoded JPEG image data in grayscale.
+///
+/// # Example
+///
+/// ```
+/// use kornia_image::Image;
+/// use kornia_io::functional as F;
+/// use std::fs;
+///
+/// let jpeg_data = fs::read("../../tests/data/dog.jpeg").unwrap();
+/// let image: Image<u8, 1> = F::decode_image_jpegturbo_gray8(&jpeg_data).unwrap();
+/// ```
+pub fn decode_image_jpegturbo_gray8(jpeg_data: &[u8]) -> Result<Image<u8, 1>, IoError> {
+    // decode the data directly from memory
+    let image: Image<u8, 1> = {
+        let mut decoder = JpegTurboDecoder::new()?;
+        decoder.decode_gray8(jpeg_data)?
+    };
+
+    Ok(image)
+}
+
+/// Decodes an image in `GRAY8` format directly from bytes.
+///
+/// The method tries to decode image data from any format supported by the image crate
+/// and converts it to grayscale.
+///
+/// # Arguments
+///
+/// * `image_data` - The raw image data as a byte array.
+/// * `format_hint` - Optional string hint about the image format (e.g., "png", "jpeg", etc.).
+///                  This helps the decoder determine the image format if it's ambiguous.
+///
+/// # Returns
+///
+/// An image containing the decoded image data in GRAY8 format.
+///
+/// # Example
+///
+/// ```
+/// use kornia_image::Image;
+/// use kornia_io::functional as F;
+/// use std::fs;
+///
+/// let image_data = fs::read("../../tests/data/dog.jpeg").unwrap();
+/// let image: Image<u8, 1> = F::decode_image_bytes_gray8(&image_data, Some("jpeg")).unwrap();
+/// ```
+pub fn decode_image_bytes_gray8(image_data: &[u8], format_hint: Option<&str>) -> Result<Image<u8, 1>, IoError> {
```

Comment:
```suggestion
pub fn decode_image_bytes_jpg_gray8(image_data: &[u8], format_hint: Option<&str>) -> Result<Image<u8, 1>, IoError> {
```
let's go first through the path to have explicit functions